src/SDL12_compat.c

```diff
@@ -4124,7 +4124,8 @@ static void
 HandleInputGrab(SDL12_GrabMode mode)
 {
     /* SDL 1.2 always grabbed input if the video mode was fullscreen. */
-    const SDL_bool wantgrab = ((VideoSurface12->flags & SDL12_FULLSCREEN) || (mode == SDL12_GRAB_ON)) ? SDL_TRUE : SDL_FALSE;
+    const SDL_bool isfullscreen = (VideoSurface12) ? (VideoSurface12->flags & SDL12_FULLSCREEN) : SDL_FALSE;
```

Comment:
I suggest changing this to 
`    const SDL_bool isfullscreen = (VideoSurface12 && VideoSurface12->flags & SDL12_FULLSCREEN) ? SDL_TRUE : SDL_FALSE;`
for better compatibility (e.g. for c++ compilers.)

Reply:
Done.